File: Back-restaurante/src/entities/IngredienteMenu.js
const { EntitySchema } = require("typeorm");

module.exports = new EntitySchema({
  name: "IngredienteMenu",
  tableName: "ingrediente_menu",
  columns: {
    id_platillo_fk: {
      primary: true,
      type: "number",
    },
    id_ingrediente_fk: {
      primary: true,
      type: "number",
    },
    cantidad_utilizada: {
      type: "number",
      nullable: true,
    },
  },
  relations: {
    platillo: {
      type: "many-to-one",
      target: "Menu",
      joinColumn: {
        name: "id_platillo_fk",
        referencedColumnName: "id_platillo"
      }
    },
    ingrediente: {
      type: "many-to-one",
      target: "StoreIngrediente",
      joinColumn: {
        name: "id_ingrediente_fk",
        referencedColumnName: "id_ingrediente"
      }
    }
  }
});

File: backend/backend/src/main/java/restaurante/backend/IngredienteMenu/domain/IngredienteMenu.java
package restaurante.backend.IngredienteMenu.domain;

import jakarta.persistence.*;
import lombok.Data;
import restaurante.backend.Menu.domain.Menu;
import restaurante.backend.StoreIngrediente.domain.StoreIngrediente;
@Entity
@Table(name = "ingrediente_menu")
@Data
public class IngredienteMenu {
    @EmbeddedId
    private IngredienteMenuPK id;

    @ManyToOne
    @MapsId("idPlatilloFk")
    @JoinColumn(name = "id_platillo_fk", nullable = false)
    private Menu platillo;

    @ManyToOne
    @MapsId("idIngredienteFk")
    @JoinColumn(name = "id_ingrediente_fk", nullable = false)
    private StoreIngrediente ingrediente;

    @Column(name = "cantidad_utilizada")
    private double cantidadUtilizada;
}
